test/functional/feature_assumeutxo.py

```diff
@@ -644,7 +644,11 @@ def check_tx_counts(final: bool) -> None:
         for i in (0, 1):
             n = self.nodes[i]
             self.log.info(f"Restarting node {i} to ensure (Check|Load)BlockIndex passes")
-            self.restart_node(i, extra_args=self.extra_args[i])
+            # Ensure the assumutxo node has cleaned up the background chainstate
+            cleanup_msg = ["cleaning up unneeded background chainstate"] if i != 0 else []
+            with n.assert_debug_log(cleanup_msg):
+                self.restart_node(i, extra_args=self.extra_args[i])
+            assert not (n.datadir_path / "regtest" / "chainstate_snapshot").exists()
 
```

Comment:
Ensures proper cleanup of background chainstate after `assumeutxo` node restart to prevent disk space leaks